src/Microsoft.OpenApi/Models/OpenApiParameter.cs

```diff
@@ -257,6 +257,12 @@ public void SerializeAsV2WithoutReference(IOpenApiWriter writer)
             {
                 writer.WriteOptionalObject(OpenApiConstants.Schema, Schema, (w, s) => s.SerializeAsV2(w));
             }
+            // In V2 parameter's type can't be a reference to a custom object schema or can't be of type object
+            // So in that case map the type as string.
+            else if (Schema?.Reference?.Type == ReferenceType.Schema || Schema?.Type == "object")
```

Comment:
The referenced schema might actually be a primitive schema, like string or integer.  I would change the condition to  `else if (Schema?.UnresolvedReference == false && Schema?.Type == "object") ` If someone wants to output the referenced schema as `type: string` they will have to do the resolution.

Reply:
ok will fix it.

---
In reply to: [175582906](https://github.com/Microsoft/OpenAPI.NET/pull/231#discussion_r175582906) [](ancestors = 175582906)